control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3980,7 +3979,7 @@ func (r *HostedControlPlaneReconciler) reconcileOperatorLifecycleManager(ctx con
 
 			var getCatalogImagesErr error
 			olmCatalogImagesOnce.Do(func() {
-				catalogImages, err = olm.GetCatalogImages(ctx, *hcp, pullSecret.Data[corev1.DockerConfigJsonKey], registryclient.GetListDigest, r.ImageMetadataProvider)
+				catalogImages, err = olm.GetCatalogImages(ctx, *hcp, pullSecret.Data[corev1.DockerConfigJsonKey], r.ImageMetadataProvider.GetDigest, r.ImageMetadataProvider, isImageRegistryOverrides)
```

Comment:
we are already passing `ImageMetadataProvider`, why do we need to pass `ImageMetadataProvider.GetDigest` explicitly? I think we can drop it.

Reply:
It so that we can mock it for the test, https://github.com/openshift/hypershift/pull/5551/files#diff-5846297ea35ea4e63b978b59b6773f58ff8d63668d14648f764ec523b92a5554R149

Reply:
that test is in the HCCO and doesn't seem to be calling `olm.GetCatalogImages` func. Even if it does you also pass a mock imageMetaDataProvider there, which will container the mock `GetDigest` func

Reply:
yeah it tests the reconcileOLM which uses NewOperatorLifecycleManagerParams which uses the olm.getCatalogImages.

And the way the fakeImageMetaDataProvider is designed still relies on a image being available in a repo so I couldn't mock the image in a registry for the getRepoSetup() func which causes this unit test to fail when using the fakeImageMetaDataProvider.

Reply:
you could just create a new fakeImageMetaDataProvider and use it for that test, no need to reuse the existing one.

Reply:
this code is already confusing with too many things passed around from different places.

Reply:
I created another fakeRegistryClientProvider to mock the functions for the HCCO where its needed in the olm reconcile.

Reply:
great, thanks. This is not needed now right? https://github.com/openshift/hypershift/pull/5551/files#diff-2cc6d15e7cd4dc8dbc5d56444fc81df070bc6749714b11e8b6646a50d79003a0R389
```
type GetDigestFN = func(ctx context.Context, imageRef string, pullSecret []byte) (digest.Digest, *reference.DockerImageReference, error)
```